### ISSUE TITLE
Non-english projects csv export via scheduled job fails with encoding error

### ISSUE THREAD

**Issue by dchhabda** [Scifabric/pybossa#1654]
Hi, Looks like scheduled export project fails for projects having tasks in non-english characters. I verified that this can be fixed with encoding to utf-8 during export to csv before the data is flatten.

Here is the error stack trace,
  File "/home/ubuntu/pybossa/pybossa/exporter/__init__.py", line 85, in _get_data
    tmp.append(flatten(cleaned))
  File "/home/ubuntu/pybossa/env/src/flatten-json/flatten_json.py", line 58, in flatten
    _flatten(nested_dict, None)
  File "/home/ubuntu/pybossa/env/src/flatten-json/flatten_json.py", line 51, in _flatten
    _flatten(object_[object_key], _construct_key(key, separator, object_key))
  File "/home/ubuntu/pybossa/env/src/flatten-json/flatten_json.py", line 51, in _flatten
    _flatten(object_[object_key], _construct_key(key, separator, object_key))
  File "/home/ubuntu/pybossa/env/src/flatten-json/flatten_json.py", line 18, in _construct_key
    return "{}{}{}".format(previous_key, separator, new_key)
UnicodeEncodeError: 'ascii' codec can't encode character u'\ufeff' in position 0: ordinal not in range(128)

**Comment by teleyinex** [comment]
Hi,

We've been chasing this issue, but we didn't have the full solution yet. We thought it was a problem with the flatten library, but now this has been fixed, so the issue remains in the export.

Where did you exactly fixed it? Would you mind to send us a patch?

**Comment by teleyinex** [comment]
Just found the way to make the tests fail :-) Now fixing it!

**Comment by teleyinex** [comment]
It should be solved in #1655

**Comment by dchhabda** [comment]
With existing flatten library, the fix that I'd was to encode dict values adding lines after [this](https://github.com/Scifabric/pybossa/blob/master/pybossa/exporter/__init__.py#L55) 
inf = {k: unicode(v).encode("utf-8") for k,v in inf.iteritems()}

and [this](https://github.com/Scifabric/pybossa/blob/master/pybossa/exporter/__init__.py#L72)
cleaned = {k: unicode(v).encode("utf-8") for k,v in cleaned.iteritems()}

**Comment by teleyinex** [comment]
Yes, we tested this as well, but at the end we wanted to fix the original problem: the flatten library. We patched it locally, we sent the patch upstream (and it has been accepted by the owner) so in the coming weeks they will publish it and we can install it directly from pip.

Also, we had an issue with StringIO in our tests. StringIO does not handle UTF8 at all, so it was making them fail. Now it's solved as well, ensuring that PYBOSSA always exports files in utf8, so this should be fixed for ever :-)

Actually, we deployed yesterday this in Crowdcrafting and all utf8 projects with keys in utf8 are now no giving errors. :-)
